# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.h
# Commit: 95679f6ce254
# Full Hash: 95679f6ce2544b21ec9cfd0d19961058d60155d6
# Author: Frederic Wang <fred.wang@free.fr>
# Date: 2023-11-27 17:00:06
# Regressor Bug: 1807253
# File Overlap Count: 1
# Description:
#   Bug 1807253 - unreliable timing for content-visibility:auto, r=emilio
#   
#   Rewrite implementation of content-visibility: auto as defined in
#   https://github.com/w3c/csswg-drafts/issues/8542
#   
# ==============================================================================

diff -r 9ea04a2cc829 -r 95679f6ce254 dom/base/Document.h
--- a/dom/base/Document.h	Mon Nov 27 08:00:41 2023 +0000
+++ b/dom/base/Document.h	Mon Nov 27 08:26:46 2023 +0000
@@ -3709,13 +3709,6 @@
   DOMIntersectionObserver* GetLazyLoadObserver() { return mLazyLoadObserver; }
   DOMIntersectionObserver& EnsureLazyLoadObserver();
 
-  DOMIntersectionObserver* GetContentVisibilityObserver() const {
-    return mContentVisibilityObserver;
-  }
-  DOMIntersectionObserver& EnsureContentVisibilityObserver();
-  void ObserveForContentVisibility(Element&);
-  void UnobserveForContentVisibility(Element&);
-
   ResizeObserver* GetLastRememberedSizeObserver() {
     return mLastRememberedSizeObserver;
   }
@@ -3762,7 +3755,16 @@
    * Returns whether there is any ResizeObserver that has skipped observations.
    */
   bool HasAnySkippedResizeObservations() const;
-  MOZ_CAN_RUN_SCRIPT void NotifyResizeObservers();
+  /**
+   * Returns whether the document contains any content-visibility: auto element.
+   */
+  bool HasContentVisibilityAutoElements() const;
+  /**
+   * Determine proximity to viewport for content-visibility: auto elements and
+   * notify resize observers.
+   */
+  MOZ_CAN_RUN_SCRIPT void
+  DetermineProximityToViewportAndNotifyResizeObservers();
 
   // Getter for PermissionDelegateHandler. Performs lazy initialization.
   PermissionDelegateHandler* GetPermissionDelegateHandler();
@@ -5118,10 +5120,6 @@
 
   RefPtr<DOMIntersectionObserver> mLazyLoadObserver;
 
-  // Used for detecting when `content-visibility: auto` elements are near
-  // or far from the viewport.
-  RefPtr<DOMIntersectionObserver> mContentVisibilityObserver;
-
   // ResizeObserver for storing and removing the last remembered size.
   // @see {@link https://drafts.csswg.org/css-sizing-4/#last-remembered}
   RefPtr<ResizeObserver> mLastRememberedSizeObserver;